serverless/serverless
# S3 Block Public Access

## What did you implement

* Add a new boolean property named `blockPublicAccess`to the
  `deploymentBucket` config object for the AWS provider.
* The property defaults to `false` to retain backwords
  compatibility with existing infrastructure.
* When `true`, the configuration property will enable the
  [S3 Block Public Access feature][1]. This will prevent all future
  policies or ACLs that may grant public access to the
  deployment bucket.
* Update the serverless.yml reference docs to reflect the
  addition.

[1]: https://aws.amazon.com/blogs/aws/amazon-s3-block-public-access-another-layer-of-protection-for-your-accounts-and-buckets/

Closes https://github.com/serverless/serverless/issues/6706

## How can we verify it

Add `blockPublicAccess: true` to the `deploymentBucket` configuration object, then deploy. 

Ensure that the resulting CF template has the following properties in the `ServerlessDeploymentBucket` resource:

```json
{
  "PublicAccessBlockConfiguration": {
     "BlockPublicAcls": true,
     "BlockPublicPolicy": true,
     "IgnorePublicAcls": true,
     "RestrictPublicBuckets": true
   }
}
```


**_Is this ready for review?:_** Yes
**_Is it a breaking change?:_** No
